src/test_typing_extensions.py

```diff
@@ -7246,6 +7286,196 @@ def test_getitem(self):
         self.assertEqual(get_args(fully_subscripted), (Iterable[float],))
         self.assertIs(get_origin(fully_subscripted), ListOrSetT)
 
+        # Test ParamSpec and Ellipsis
+        P = ParamSpec('P')
+        CallableP = TypeAliasType("CallableP", Callable[P, Any], type_params=(P,))
+        # () -> Any
+        callable_no_arg = CallableP[[]]
+        self.assertEqual(get_args(callable_no_arg), ([],))
+        # (int) -> Any
+        callable_arg = CallableP[int]
+        self.assertEqual(get_args(callable_arg), (int,))
+
+        callable_arg_list = CallableP[[int]]
+        self.assertEqual(get_args(callable_arg_list), ([int],))
+
+        # (int, int) -> Any
+        callable_arg2 = CallableP[int, int]
+        self.assertEqual(get_args(callable_arg2), (int, int,))
+
+        callable_arg2_list = CallableP[[int, int]]
+        self.assertEqual(get_args(callable_arg2_list), ([int, int],))
+        # (...) -> Any
+        callable_ellipsis = CallableP[...]
+        self.assertEqual(get_args(callable_ellipsis), (...,))
+
+        callable_ellipsis2 = CallableP[(...,)]
+        self.assertEqual(callable_ellipsis, callable_ellipsis2)
+        # (int, ...) -> Any
+        callable_arg_more = CallableP[[int, ...]]
+        self.assertEqual(get_args(callable_arg_more), ([int, ...],))
+        # (T) -> Any
+        callable_generic_raw = CallableP[T]
+        self.assertEqual(get_args(callable_generic_raw), (T,))
+        self.assertEqual(callable_generic_raw.__parameters__, (T,))
+
+        # Usage with Concatenate
+        callable_concat = CallableP[Concatenate[int, P]]
+        self.assertEqual(callable_concat.__parameters__, (P,))
+        if TYPING_3_11_0:
+            self.assertEqual(get_args(callable_concat), (Concatenate[int, P],))
+            concat_usage = callable_concat[str]
+            self.assertEqual(get_args(concat_usage), ((int, str),))
+            self.assertEqual(concat_usage, callable_concat[[str]])
```

Comment:
Since using `types.GenericAlias` these tests became much clearer. Given that `Concatenate` might need some special treatmeant in `TypeAliasType.__getitem__`  (currently done by `isinstance(param, list)`) and this is the only test where this is looked into with a bit more detail, I think it should probaly stay.